toolchain/lower/file_context.cpp

```diff
@@ -249,6 +249,9 @@ auto FileContext::BuildFunctionDecl(SemIR::FunctionId function_id)
       llvm::Function::Create(function_type, llvm::Function::ExternalLinkage,
                              mangled_name, llvm_module());
 
+  CARBON_CHECK(llvm_function->getName() == mangled_name,
+               "Do not rely on llvm::Function overload renaming");
```

Comment:
I think it'd be useful to include `mangled_name` in the diagnostic here.

Reply:
Oh, didn't know CARBON_CHECK could have a parameterized message. Done that